src/onegov/org/views/people.py

```diff
@@ -34,10 +40,21 @@ def get_items(self):
 @OrgApp.html(model=Person, template='person.pt', permission=Public)
 def view_person(self, request, layout=None):
 
-    pages = request.session.query(Topic)
-    pages = pages.filter(Topic.people is not None).all()
-    org_to_func = person_functions_by_organization(self, pages, request)
+    def visit_topics_with_people(
+        pages: 'Iterable[Page]',
+        root_id: int | None = None
+    ) -> 'Iterator[Topic]':
+        for page in pages:
+
+            if root_id is None:
+                root_id = page.id
 
+            if isinstance(page, Topic) and page.people:
```

Comment:
Since `page.people` does a bunch of work that has to be repeated every time the property is accessed it might be better to check for `page.content.get('people')` here. Since `person_functions_by_organization` accesses this attribute again.

Reply:
ah yes, we can save ourselves this call.